src/rgw/rgw_gc.cc

```diff
@@ -151,6 +151,7 @@ class RGWGCIOManager {
 
   deque<IO> ios;
   vector<std::vector<string> > remove_tags;
+  vector<map<string, size_t> > tag_io_size;
```

Comment:
Please consider adding comment (with appropriate formatting):
`// tracks the number of remaining shadow objects for a given tag in order to only remove the tag once all shadow objects have themselves been removed`